packages/terra-framework-docs/src/terra-dev-site/doc/worklist-data-grid/Examples.2/PinnedColumnsWithRowSelection.jsx

```diff
@@ -0,0 +1,146 @@
+    { id: 'Column-8', displayName: 'Medication History' },
+    { id: 'Column-9', displayName: 'My Relationship' },
+    { id: 'Column-10', displayName: 'Not Selectable', isSelectable: false },
+  ],
+  rows: [
+    {
+      id: '1',
+      cells: [
+        { content: 'Fleck, Arthur' },
+        { content: '1007-MTN' },
+        { content: 'Unstable' },
+        { content: 'Inpatient, 2 months' },
+        { content: '' },
+        { content: 'Quinzell, Harleen' },
+        { content: '' },
+        { isMasked: true },
+        { isMasked: true },
+        { content: 'Admitting Physician' },
+        { content: '', isSelectable: false },
+      ],
+    },
+    {
+      id: '2',
+      cells: [
+        { content: 'Wayne, Bruce' },
+        { content: '1007-MTN-DR' },
+        { content: 'Stable' },
+        { content: 'Outpatient, 2 days' },
+        { content: 'Phytochemicals' },
+        { content: 'Grayson, Richard' },
+        { content: '' },
+        { content: '' },
+        { isMasked: true },
+        { content: 'Admitting Physician' },
+        { content: '', isSelectable: false },
+      ],
+    },
+  ],
+};
+
+const RowSelection = () => {
+  const rowSelectionModeRef = useRef();
+  const rowHeaderIndex = 0;
+  const { cols, rows } = gridDataJSON;
+  const [selectedRows, setSelectedRows] = useState([]);
+  const [hasSelectableRows, setHasSelectableRows] = useState(false);
+
+  const determineSelectedRows = (allRowsSelected, userSelectedRow) => {
+    if (!userSelectedRow) {
+      return [];
+    }
+
+    let remainingSelectedRow = [];
+    if (allRowsSelected) {
+      remainingSelectedRow = userSelectedRow;
+    } else if (selectedRows.includes(userSelectedRow[0])) {
+      // Row Deselected so remove this rowId.
+      remainingSelectedRow = selectedRows.filter(e => (e !== userSelectedRow[0]));
+    } else {
+      // Row Selected so add this rowId.
+      remainingSelectedRow = remainingSelectedRow.concat(selectedRows);
+      remainingSelectedRow.push(userSelectedRow[0]);
+    }
+    return remainingSelectedRow;
+  };
+
+  const clearRowSelection = () => {
+    // eslint-disable-next-line no-param-reassign
+    rows.forEach(r => { if (r.isSelected) { r.isSelected = false; } });
+    setSelectedRows([]);
+  };
+
+  const disableSelectableRows = () => {
+    rowSelectionModeRef.current.checked = false;
+    setHasSelectableRows(false);
+    clearRowSelection();
+  };
+
+  const onRowSelectionModeChange = (event) => {
+    if (!event.target.checked) {
+      clearRowSelection();
+    }
+    setHasSelectableRows(event.target.checked);
+  };
+
+  return (
+    <React.Fragment>
+      <div>
+        <label htmlFor="rowSelectionMode"><b>Row Selection Mode</b></label>
+        <input
+          id="rowSelectionMode"
+          type="checkbox"
+          ref={rowSelectionModeRef}
+          onChange={onRowSelectionModeChange}
+        />
+      </div>
+      <WorklistDataGrid
+        id="default-terra-worklist-data-grid"
```

Comment:
This example should have its own unique identifier.

Reply:
Updated: [beaa707](https://github.com/cerner/terra-framework/pull/1710/commits/beaa70782cbdf8729ccedd9c1bd337c9dab88314)